internal/core/metadata/rest_provisionwatcher.go

```diff
@@ -371,15 +397,23 @@ func updateProvisionWatcherFields(from models.ProvisionWatcher, to *models.Provi
 }
 
 // Notify the associated device services for the provision watcher
-func notifyProvisionWatcherAssociates(pw models.ProvisionWatcher, action string) error {
+func notifyProvisionWatcherAssociates(
+	pw models.ProvisionWatcher,
+	action string,
+	loggingClient logger.LoggingClient) error {
+
 	// Get the device service for the provision watcher
 	ds, err := dbClient.GetDeviceServiceById(pw.Service.Id)
 	if err != nil {
 		return err
 	}
 
 	// Notify the service
-	if err = notifyAssociates([]models.DeviceService{ds}, pw.Id, action, models.PROVISIONWATCHER); err != nil {
+	if err = notifyAssociates([]models.DeviceService{ds},
+		pw.Id,
+		action,
+		models.PROVISIONWATCHER,
+		loggingClient); err != nil {
```

Comment:
Preference would be to do the assignment before the if statement.  This results in a single line if statement and is IMO more readable/parseable.

Reply:
Not updated.  Expecting this:

```
err = notifyAssociates([]models.DeviceService{ds}, pw.Id, action, models.PROVISIONWATCHER, loggingClient);
if  err != nil {
		return err
}```